### ISSUE TITLE
Incorrect RefSeq transcript shown

### ISSUE THREAD

**Issue by drtconway** [broadinstitute/seqr#3325]
Consider the variant shown at https://seqr.broadinstitute.org/variant_search/variant/1-175362686-C-G/family/F031959_demofamily1

If you click on the variant consequence "missense variant", you get the page showing the transcripts and associated consequence. In this particular instance you also get shown the RefSeq ID `NM_001328635.1`. However, if you compose the accession with the `c.` HGVS description to form `NM_001328635.1:c.2831G>C`, the resulting variant is invalid (see the mutalyzer result: https://mutalyzer.nl/normalizer/NM_001328635.1:c.2831G%3EC).

The problem is that the RefSeq transcript being displayed is incorrect. The correct transcript is `NM_003285.3`, and if you use that instead, the HGVS is correct.

I have observed the problem on this gene (TNR), but I don't know if other genes are affected, though I'm guessing there are others.

It appears that the Ensembl to RefSeq mappings are flawed - the problem appears on the CPG instance, and our local instance at MCRI.

With respect to severity, it's hard to say how big the problem is. My concern would be that if someone composes the HGVSc in the way I described above and puts it in a publication, it won't be hard for someone using the variant to discover that it is wrong, but it will be hard for them to work out the correct transcript. Or worse - the HGVSc actually validates but does not describe the intended variant.

**Comment by drtconway** [comment]
It looks like the problem may be related to how the mappings are derived. It looks like they are taken from dbNSFP. Unfortunately dbNSFP only associates the RefSeq IDs to genes, not to transcripts. In this instance, dbNFSP has both IDs: `NM_003285;NM_001328635` and it looks like Seqr is showing the second one (which is lexicographically first?).

Three possible ways forward occur to me immediately.

The simpler, but not as good way, would be to move the `RefSeq:` block to the top "gene" part of the Transcripts panel, and show all the RefSeq IDs, not just one of them. This doesn't require any data change in the back end (good!), and avoids misleadingly giving the impression that the displayed RefSeq ID associates to the particular transcript (good!). It doesn't, however, give the user the correct RefSeq ID (if there is one) for the transcript used to characterise the consequence of the variant (unfortunate!).

A more difficult change would be to use the MANE data to give the RefSeq ID for the MANE transcript only. I don't know much about the back end, but I assume this would require a database change (bad!), but would show correct ENST-RefSeq mappings (good!), though not for all possible transcripts for which an Ensembl-RefSeq mapping exsists (unfortunate!).

A more complete resolution would be to use biomart to get a "complete" Ensembl-RefSeq mapping, and to display that information in the Transcripts panel. I assume this would require backend database changes (bad!), and reworking of the Transcript panel code (bad!), but would display accurate (good!) and complete (good!) information.

The bottom line is that it would be good to avoid showing misleading information, and if possible to show helpful information.

**Comment by drtconway** [comment]
Thanks Hannah.

I think the option 1 I suggested of moving the RefSeq to the Gene level in combination with your suggesting is a good short term option. I think MANE is the best medium term option.

**Comment by drtconway** [comment]
I forgot to mention: it would also clarify things if all the RefSeq accessions were displayed rather than just one. That would make it immediately clear that the displayed information should not be taken as a 1-to-1 Ensembl transcript to RefSeq mapping.

**Comment by hanars** [comment]
Moving the transcript to gene-level does not meet the needs of our core user base, who explicitly asked for the per-transcript mapping. Despite occasional inaccuracies, it is much easier for them to start with a single likely transcript and confirm it is correct, rather than a list of the potentially hundereds of transcripts in a gene

**Comment by drtconway** [comment]
I may have misinterpreted how the RefSeq information is handled. Looking at a variant in `Dock7` it is clearly showing multiple ENST-to-RefSeq mappings. However, I haven't succeeded in finding out the correct source data to try and track down why it is wrong for `TNR`. Can you give me a pointer to where that data comes from?

Obviously it's not the systematic problem that my previous analysis suggested.

**Comment by hanars** [comment]
Its from the gencdoe FTP server:
http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/gencode.v31.metadata.RefSeq.gz
